modules/embedding-microshift-into-rpm-ostree-image.adoc

```diff
@@ -0,0 +1,212 @@
+
+This procedure can be summarized as copying the repository definition file for
+your {op-system} version from the `/usr/share/osbuild-composer/repositories`
+directory into the override directory at `/etc/osbuild-composer/repositories`,
+appending new repository entries for the target CPU architecture (e.g. "x86_64")
+you are building for, and reloading the `osbuild-composer` service. The
+repository definition file to copy needs to correspond to your {op-system}
+version. For example, for {op-system} 8.7, it is called `rhel-87.json`.
+
+Add MicroShift's RPM repos by using the following command to override the
+repository definition (here shown for {op-system} 8.7):
+
+[source,terminal]
+----
+$ arch=$(uname -i) jq ".$arch +=
+    [(.$arch[0] + {
+        \"name\": \"rhocp-4.12-for-rhel-8-$arch-rpms\",
+        \"baseurl\": \"https://cdn.redhat.com/content/dist/layered/rhel8/$arch/rhocp/4.12/os\"}
+    )] +
+    [(.$arch[0] + {
+        \"name\": \"fast-datapath-for-rhel-8-$arch-rpms\",
+        \"baseurl\": \"https://cdn.redhat.com/content/dist/layered/rhel8/$arch/fast-datapath/os\"}
+    )]
+    " /usr/share/osbuild-composer/repositories/rhel-87.json | sudo tee /etc/osbuild-composer/repositories/rhel-87.json
+----
+
+Verify the entries have been added. They should look like the following:
+
+[source,terminal]
+----
+$ tail -n 16 /etc/osbuild-composer/repositories/rhel-87.json
+    {
+      "name": "rhocp-4.12-for-rhel-8-x86_64-rpms",
+      "baseurl": "https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/rhocp/4.12/os",
+      "gpgkey": "-----BEGIN PGP PUBLIC KEY BLOCK-----...-----END PGP PUBLIC KEY BLOCK-----\n",
+      "rhsm": true,
+      "check_gpg": true
+    },
+    {
+      "name": "fast-datapath-for-rhel-8-x86_64-rpms",
+      "baseurl": "https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/fast-datapath/os",
+      "gpgkey": "-----BEGIN PGP PUBLIC KEY BLOCK----- [...] -----END PGP PUBLIC KEY BLOCK-----\n",
+      "rhsm": true,
+      "check_gpg": true
+    }
+  ]
+}
+----
+
+Restart the `osbuild-composer.service` to load the override repository definition:
+
+[source,terminal]
+----
+$ sudo systemctl restart osbuild-composer.service
+----
+
+== Adding MicroShift to an Image Builder blueprint
+=== Adding the MicroShift service
+To build an {op-system-ostree} image with MicroShift, simply add the MicroShift
+RPM package to a blueprint and enable the MicroShift service. A minimal Image
+Builder blueprint would look as follows:
+
+[source,terminal]
+----
+name = "minimal-microshift"
+
+description = ""
+version = "0.0.1"
+modules = []
+groups = []
+
+[[packages]]
+name = "microshift"
+version = "{ocp-version}.0-1"
+
+[[packages]]
+name = "redhat-release"
+version = "*"
+
+[customizations.services]
+enabled = ["microshift"]
+----
+
+If your use case requires running the `oc` CLI tool locally on your MicroShift
```

Comment:
They would add a packages stanza for the new package, using the same style as above. We could give that example separately but it seems like that would complicate the doc here. Maybe we should just always include it in the example above instead of introducing the conditional?

Reply:
I fear if we include it, users may interpret this as recommendation / requirement. Also, it’d no longer be a *minimal* example.

Reply:
OK, I'm looking for ways to keep the doc simple. I think @ShaunaDiaz point is that these 2 sentences are not enough by themselves. So do we want a completely separate example procedure for the remote git repo case (I could see how an end-to-end example of that would be helpful, but also a lot to write)? If we want it, I think I'd put off trying to write that until 4.13 opens.